### ISSUE TITLE
error opening modeler

### ISSUE THREAD

**Issue by yeufui** [bpmn-io/vs-code-bpmn-io#76]
![vs-code-info](https://user-images.githubusercontent.com/22479145/84756716-bd89e280-aff5-11ea-80f9-d4226ad80294.png)

![error-prompt](https://user-images.githubusercontent.com/22479145/84756873-f75ae900-aff5-11ea-896d-2f8f3aeee640.png)

__Describe the Bug__

cannot open the modeler in vscode

__Steps to Reproduce__

1. click `ctrl + shift + p` to open command pallete
2. select `Open BPMN Modeler`
3. error prompt as screen attached

__Expected Behavior__

__Environment__

- Host (Browser/Node version), refer attached screens
- OS: Win 10

**Comment by pinussilvestrus** [comment]
Which version of the extension are you using?

**Comment by pinussilvestrus** [comment]
Okay, I was able to reproduce it on my machine with the newest version of the extension. Thanks for reporting, we'll have a look!

Instead of using the command from the palette, you can easily open the Modeler from the editor bar

![image](https://user-images.githubusercontent.com/9433996/84759740-6cfb9080-afc7-11ea-8b8b-68c754422c10.png)

**Comment by pinussilvestrus** [comment]
The error relates to the fact the [file URI is undefined when executing it via the command palette](https://github.com/bpmn-io/vs-code-bpmn-io/blob/master/src/extension.ts#L169)

**Comment by programmiererei** [comment]
I still have this error with vscode:
Version: 1.46.1 (system setup)
Commit: cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
Datum: 2020-06-17T21:13:20.174Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
Betriebssystem: Windows_NT x64 10.0.18363

and version 0.8.0

**Comment by pinussilvestrus** [comment]
We have unreleased features which fix this issue: https://github.com/bpmn-io/vs-code-bpmn-io/compare/v0.8.0...master

Would you say this error is bothering you that much so the fix needs to be released shortly?

**Comment by programmiererei** [comment]
Hi,

 

thanks for the answer.

 

I started debugging with 0.8.0…master.

 

I opened the file c:\Users\ralfa\Source\repos\his-docu\test.bpmn <file:///c:/Users/ralfa/Source/repos/his-docu/test.bpmn>  and pressed the button to open BPMN Explorer.

 

The problem is in

 

public provideTextDocumentContent(localResource: vscode.Uri, webview: vscode.Webview): string {

 

    const webViewUri = webview.asWebviewUri(localResource);

 

    const docPath = webViewUri.fsPath;

 

    const contents = fs.readFileSync(docPath, { encoding: 'utf8' });

 

    const builder = new BpmnModelerBuilder(contents, {

      modelerDistro: this.getUri(webview, 'node_modules', 'bpmn-js', 'dist', 'bpmn-modeler.development.js'),

      diagramStyles: this.getUri(webview, 'node_modules', 'bpmn-js', 'dist', 'assets', 'diagram-js.css'),

      bpmnFont: this.getUri(webview, 'node_modules', 'bpmn-js', 'dist', 'assets', 'bpmn-font', 'css', 'bpmn.css'),

      modelerStyles: this.getUri(webview, 'out', 'assets', 'modeler.css'),

      resourceUri: localResource

    });

 

    return builder.buildModelerView();

  }

 

The call to fs.readFileSync fails for docPath: \\\c:\Users\ralfa\Source\repos\his-docu\test.bpmn <file:///c:/Users/ralfa/Source/repos/his-docu/test.bpmn> 

 

The exception leads to the catch clause in _executeContributedCommand

 

Grüße

 

Ralf Albrecht

 

 

programmiererei nicol gmbh

rotlintstr. 14  …  60316 frankfurt

fon 069.437152  …  fax 069.94414224

 <mailto:albrecht@programmiererei.de> albrecht@programmiererei.de

…

sitz der gmbh:  frankfurt hrb 51334

geschäftsführer: ralf albrecht

 

Von: Niklas Kiefer <notifications@github.com> 
Gesendet: Dienstag, 7. Juli 2020 08:17
An: bpmn-io/vs-code-bpmn-io <vs-code-bpmn-io@noreply.github.com>
Cc: programmiererei <programmiererei@web.de>; Comment <comment@noreply.github.com>
Betreff: Re: [bpmn-io/vs-code-bpmn-io] error opening modeler (#76)

 

We have unreleased features which fix this issue:  <https://github.com/bpmn-io/vs-code-bpmn-io/compare/v0.8.0...master> v0.8.0...master

Would you say this error is bothering you that much so the fix needs to be released shortly?

**Comment by pinussilvestrus** [comment]
Yes, we know. As I mentioned, we already fixed it with https://github.com/bpmn-io/vs-code-bpmn-io/commit/ceaa31e2bf14bc544f698bd245c2a107c2fc8232, but the changes got not released yet.

**Comment by programmiererei** [comment]
Hi,

 

but thats the code version ( <https://github.com/bpmn-io/vs-code-bpmn-io/commit/ceaa31e2bf14bc544f698bd245c2a107c2fc8232> ceaa31e) I used for debugging:

 



 

 

Grüße

 

Ralf Albrecht

 

 

programmiererei nicol gmbh

rotlintstr. 14  …  60316 frankfurt

fon 069.437152  …  fax 069.94414224

 <mailto:albrecht@programmiererei.de> albrecht@programmiererei.de

…

sitz der gmbh:  frankfurt hrb 51334

geschäftsführer: ralf albrecht

 

Von: Niklas Kiefer <notifications@github.com> 
Gesendet: Dienstag, 7. Juli 2020 11:21
An: bpmn-io/vs-code-bpmn-io <vs-code-bpmn-io@noreply.github.com>
Cc: programmiererei <programmiererei@web.de>; Comment <comment@noreply.github.com>
Betreff: Re: [bpmn-io/vs-code-bpmn-io] error opening modeler (#76)

 

Yes, we know. As I mentioned, we already fixed it with  <https://github.com/bpmn-io/vs-code-bpmn-io/commit/ceaa31e2bf14bc544f698bd245c2a107c2fc8232> ceaa31e, but the changes got not released yet.

**Comment by pinussilvestrus** [comment]
Okay I see, this seems to be another issue then, since it is not related 100% to the triggering from the command palette but also from the editor toolbar action.

Mind you creating [a Bug Repot](https://github.com/bpmn-io/vs-code-bpmn-io/issues/new?labels=bug&template=BUG_REPORT.md) so we can handle it separately?

**Comment by programmiererei** [comment]
Hi,

 

please have a look at „Error: Running the contributed command #44“.

 

I think it is the same issue.

 

 

Grüße

 

Ralf Albrecht

 

 

programmiererei nicol gmbh

rotlintstr. 14  …  60316 frankfurt

fon 069.437152  …  fax 069.94414224

 <mailto:albrecht@programmiererei.de> albrecht@programmiererei.de

…

sitz der gmbh:  frankfurt hrb 51334

geschäftsführer: ralf albrecht

 

Von: Niklas Kiefer <notifications@github.com> 
Gesendet: Dienstag, 7. Juli 2020 11:58
An: bpmn-io/vs-code-bpmn-io <vs-code-bpmn-io@noreply.github.com>
Cc: programmiererei <programmiererei@web.de>; Comment <comment@noreply.github.com>
Betreff: Re: [bpmn-io/vs-code-bpmn-io] error opening modeler (#76)

 

Okay I see, this seems to be another issue then, since it is not related 100% to the triggering from the command palette but also from the editor toolbar action.

Mind you creating a Bug Repot <https://github.com/bpmn-io/vs-code-bpmn-io/issues/new?labels=bug&template=BUG_REPORT.md>  so we can handle it separately?

**Comment by pinussilvestrus** [comment]
Also, this issue is closed. When we want to investigate any issue, we should make sure we have a _proper_ description of the problem and a step by step guidance how to reproduce it. 

Even if we closed a similar issue back in time, it makes sense to have a _new_ issue to understand what the problem is, _why_ it popped up again, and what we can do against it. We should not re-open issues that we thought we fixed a later time ago.

**Comment by programmiererei** [comment]
OK. I’m new to this, sorry. I will open a new issue.

 

 

Grüße

 

Ralf Albrecht

 

 

programmiererei nicol gmbh

rotlintstr. 14  …  60316 frankfurt

fon 069.437152  …  fax 069.94414224

 <mailto:albrecht@programmiererei.de> albrecht@programmiererei.de

…

sitz der gmbh:  frankfurt hrb 51334

geschäftsführer: ralf albrecht

 

Von: Niklas Kiefer <notifications@github.com> 
Gesendet: Dienstag, 7. Juli 2020 15:10
An: bpmn-io/vs-code-bpmn-io <vs-code-bpmn-io@noreply.github.com>
Cc: programmiererei <programmiererei@web.de>; Comment <comment@noreply.github.com>
Betreff: Re: [bpmn-io/vs-code-bpmn-io] error opening modeler (#76)

 

Also, this issue is closed. When we want to investigate any issue, we should make sure we have a proper description of the problem and a step by step guidance how to reproduce it.

Even if we closed a similar issue back in time, it makes sense to have a new issue to understand what the problem is, why it popped up again, and what we can do against it. We should not re-open issues that we thought we fixed a later time ago.
